### DISCUSSION TITLE
Nested relatedTo params

### DISCUSSION THREAD

**Discussion by chipbell4** [craftcms/cms#8911] [Ideas]
### Description
I've been looking at the docs regarding [passing multiple relation criteria to `relatedTo`](https://craftcms.com/docs/3.x/relations.html#passing-multiple-relation-criteria) and I'm curious if this feature is supported yet:

I'd like to perform `relatedTo` queries with nested parameters, something like this:

```php
// find entries that are related to query 1 or both query 2 and 3
Entry::find()->relatedTo([
  'or',
  $query1,
  [
    'and',
    $query2,
    $query3
  ],
]);
```

I imagine this change [would occur in `ElementRelationParamParser::_subparse`](https://github.com/craftcms/cms/blob/718d223992cbcc6c2681e372008a69d9691cd5c8/src/elements/db/ElementRelationParamParser.php#L135).

**Comment by Wiejeben** [comment]
Very similar to https://github.com/craftcms/cms/issues/5733

See my comment as well https://github.com/craftcms/cms/issues/5733#issuecomment-661936108

**Comment by Wiejeben** [comment]
@brandonkelly I believe this is now fixed as well via https://github.com/craftcms/cms/issues/5733#issuecomment-800745443?

**Comment by brandonkelly** [comment]
Nah, we haven’t actually changed what `relatedTo` is capable of with 8ad4c6695b9fdb98699003b2a070b479bd18a7aa, just simplified the syntax a bit.

**Comment by joshuabaker** [comment]
Would also love this in GraphQL.

```gql
{
  categories(
    group: "facet1"
    relatedToEntries: {
      section: "articles"
      relatedToCategories: {
        group: "facet2",
        slug: "facet2slug"
      }
    }
  ) {
    ...category
  }
}
```

It would allow faceted filtering without complicated client-side logic.

**Comment by andris-sevcenko** [comment]
@joshuabaker Pretty sure this is possible already in GraphQL, though? (https://github.com/craftcms/cms/blob/develop/src/gql/base/RelationArgumentHandler.php#L91)
